Mergin/projects_manager.py

```diff
@@ -291,6 +292,10 @@ def sync_project(self, project_dir, project_name=None):
             # TODO: report success only when we have actually done anything
             msg = "Mergin project {} synchronized successfully".format(project_name)
             QMessageBox.information(None, "Project sync", msg, QMessageBox.Close)
+            # reload updated layers
+            for layer in QgsProject.instance().mapLayers().values():
+                if os.path.basename(layer.source()) in updated_sources:
+                    layer.reload()
```

Comment:
is it necessary to call `layer.reload()` like this? I thought it would be enough to just refresh map canvas...

Reply:
Unfortunately, it is required. Simple canvas refresh doesn't help here.